templates/nextflow-s3-template.yaml

```diff
@@ -0,0 +1,175 @@
+  ReadOnlyAccessArns:
+    Type: CommaDelimitedList
+    Default: "[]"
+    Description: (Optional) Grant read access to accounts, groups, and users.
+    ConstraintDescription: List of ARNs (i.e. ["arn:aws:iam::011223344556:user/jsmith", "arn:aws:iam::544332211006:user/rjones"]
+
+# Conditions
+Conditions:
+  HasBucketName: !Not [!Equals [!Ref BucketName, ""]]
+  HasReadWriteAccessArns: !Not [!Equals [!Join [",", !Ref ReadWriteAccessArns], "[]"]]
+  HasReadOnlyAccessArns: !Not [!Equals [!Join [",", !Ref ReadOnlyAccessArns], "[]"]]
+
+# Mappings
+Mappings:
+  Accounts:
+    "035458030717":
+      AdminRole: "arn:aws:iam::035458030717:role/aws-reserved/sso.amazonaws.com/AWSReservedSSO_Administrator_580e9f32ac55c4e7"
+    "728882028485":
+      AdminRole: "arn:aws:iam::728882028485:role/aws-reserved/sso.amazonaws.com/AWSReservedSSO_Administrator_3a9d61d0884be260"
+
+# Resources
+Resources:
+  Bucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !If [HasBucketName, !Ref BucketName, !Ref 'AWS::NoValue']
+      AccessControl: Private
+      OwnershipControls:
+        Rules:
+          - ObjectOwnership: BucketOwnerPreferred      
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: AES256
+
+  # Policy
+  BucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref Bucket  
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+
+          - Sid: AllowDeleteAllowModify
+            # denies all except modifying, deleting and emptying bucket
+            Effect: Deny
+            Principal: '*'
+            NotAction:
+              - "s3:Delete*"
+              - "s3:GetBucket*"
+              - "s3:ListBucket*"
+              - "s3:PutBucket*"
+            Resource:
+              - !Sub arn:aws:s3:::${Bucket}
+              - !Sub arn:aws:s3:::${Bucket}/*
+
+          - Sid: AdminFullAccess
+            # gives admin full access to bucket
+            Effect: Allow
+            Principal:
+              AWS:
+                - !Sub "arn:aws:iam::${AWS::AccountId}:root"
+                - Fn::ImportValue: !Sub "${AWS::Region}-workflows-nextflow-ci-service-account-ServiceRoleArn"
+                - !FindInMap [Accounts, !Ref AWS::AccountId, AdminRole]
+            Action:
+              - "s3:*"
+            Resource:
+              - !Sub arn:aws:s3:::${Bucket}
+              - !Sub arn:aws:s3:::${Bucket}/*
+
+          - Sid: ForgeUserFullAccess
+            # gives forge user full access to bucket
+            Effect: Allow
+            Principal:
+              AWS:
+                - Fn::ImportValue: !Sub "${AWS::Region}-nextflow-forge-service-user-ServiceUserArn"
+            Action:
+              - "s3:*"
+            Resource:
+              - !Sub arn:aws:s3:::${Bucket}
+              - !Sub arn:aws:s3:::${Bucket}/*
+
+          - !If
+            - HasReadWriteAccessArns
```

Comment:
@BrunoGrandePhD now that the !Join has been removed, should this condition be kept in?

Reply:
Nm - seems to break without